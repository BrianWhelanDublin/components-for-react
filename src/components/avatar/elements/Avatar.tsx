import React, { forwardRef } from 'react';
import { StyledAvatar } from '../styled/StyledAvatar';
import { AvatarProps } from '../types/Avatar.types';

const Avatar = forwardRef<HTMLElement, AvatarProps>(
  ({ children, size = 'medium', type = 'round', ...props }, ref) => {
    return (
      <StyledAvatar ref={ref} type={type} size={size} {...props}>
        {children}
      </StyledAvatar>
    );
  }
);

export default Avatar;
