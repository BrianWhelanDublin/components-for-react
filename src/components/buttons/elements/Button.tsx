import React from 'react';
import { forwardRef } from 'react';
import { StyledButton } from '../styled/StyledButton';
import { ButtonProps } from '../types/Button.types';

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, ...props }, ref) => {
    return (
      <StyledButton ref={ref} {...props}>
        {children}
      </StyledButton>
    );
  }
);

export default Button;
