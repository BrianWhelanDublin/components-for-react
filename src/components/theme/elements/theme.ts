import { DefaultTheme } from 'styled-components';
import {
  ThemeColors,
  ThemeOpacity,
  ThemeShadows,
  ThemeTypography,
} from '../types/theme';

export const colors: ThemeColors = {
  primary: {
    light: '#5f5fc4',
    main: '#283593',
    dark: '#001064',
  },
  secondary: {
    light: '#F9A8D4',
    main: '#DB2777',
    dark: '#831843',
  },
  tertiary: {
    light: '#6EE7B7',
    main: '#059669',
    dark: '#064E3B',
  },
  info: {
    light: '#7dd3fc',
    main: '#0284c7',
    dark: '#0c4a6e',
  },
  success: {
    light: '#86efac',
    main: '#16a34a',
    dark: '#14532d',
  },
  error: {
    light: '#fca5a5',
    main: '#dc2626',
    dark: '#7f1d1d',
  },

  warning: {
    light: '#fdba74',
    main: '#ea580c',
    dark: '#7c2d12',
  },
  singleTone: {
    white: '#ffffff',
    black: '#000000',
    lightText: '#ffffff',
    darkText: '#27272a',
  },
  grey: {
    50: '#fafafa',
    100: '#f5f5f5',
    200: '#e5e5e5',
    300: '#d4d4d4',
    400: '#a3a3a3',
    500: '#737373',
    600: '#525252',
    700: '#404040',
    800: '#262626',
    900: '#171717',
  },
  text: {
    disabled: '#d4d4d4',
    secondary: '#525252',
    primary: '#171717',
  },
};

export const typography: ThemeTypography = {
  htmlFontSize: 16,
  fontSize: 14,
  fontWeightLight: 300,
  fontWeightRegular: 400,
  fontWeightMedium: 500,
  fontWeightBold: 700,
  fontFamily: 'Helvetica',
};

export const opacity: ThemeOpacity = {
  0: 0,
  5: 0.05,
  10: 0.1,
  20: 0.2,
  25: 0.25,
  30: 0.3,
  40: 0.4,
  50: 0.5,
  60: 0.6,
  70: 0.7,
  75: 0.75,
  80: 0.8,
  90: 0.9,
  95: 0.95,
  100: 1,
};

export const shadows: ThemeShadows = {
  0: {
    shadowColor: 'rgba(0,0,0,.18)',
    shadowOffset: {
      x: 1,
      y: 1,
    },
    shadowRadius: 1,
  },
  1: {
    shadowColor: 'rgba(0,0,0,.2)',
    shadowOffset: {
      x: 1,
      y: 2,
    },
    shadowRadius: 2,
  },
  2: {
    shadowColor: 'rgba(0,0,0,.22)',
    shadowOffset: {
      x: 2,
      y: 2,
    },
    shadowRadius: 4,
  },
  3: {
    shadowColor: 'rgba(0,0,0,.23)',
    shadowOffset: {
      x: 3,
      y: 6,
    },
    shadowRadius: 6,
  },
  4: {
    shadowColor: 'rgba(0,0,0,.25)',
    shadowOffset: {
      x: 4,
      y: 8,
    },
    shadowRadius: 8,
  },
  5: {
    shadowColor: 'rgba(0,0,0,.27)',
    shadowOffset: {
      x: 5,
      y: 10,
    },
    shadowRadius: 10,
  },
  6: {
    shadowColor: 'rgba(0,0,0,.29)',
    shadowOffset: {
      x: 6,
      y: 12,
    },
    shadowRadius: 12,
  },
  7: {
    shadowColor: 'rgba(0,0,0,.3)',
    shadowOffset: {
      x: 7,
      y: 14,
    },
    shadowRadius: 14,
  },
  8: {
    shadowColor: 'rgba(0,0,0,.32)',
    shadowOffset: {
      x: 8,
      y: 16,
    },
    shadowRadius: 16,
  },
  9: {
    shadowColor: 'rgba(0,0,0,.34)',
    shadowOffset: {
      x: 9,
      y: 18,
    },
    shadowRadius: 18,
  },
};

export const defaultTheme: DefaultTheme = {
  colors,
  typography,
  opacity,
  shadows,
};
