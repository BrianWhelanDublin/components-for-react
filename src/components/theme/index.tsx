export { defaultTheme as ComponentsDefaultTheme } from './elements/theme';
export { default as ComponentsThemeProvider } from './elements/ThemeProvider';
