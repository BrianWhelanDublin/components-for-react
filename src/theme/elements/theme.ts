import {
  LibraryDefaultTheme,
  ThemeColors,
  ThemeOpacity,
  ThemeShadows,
  ThemeTypography,
} from '../types/theme.types';

export const colors: ThemeColors = {
  primary: {
    light: '#67E8F9',
    main: '#0891B2',
    dark: '#164E63',
  },
  secondary: {
    light: '#F9A8D4',
    main: '#DB2777',
    dark: '#831843',
  },
  tertiary: {
    light: '#6EE7B7',
    main: '#059669',
    dark: '#064E3B',
  },
  info: {
    light: '#7dd3fc',
    main: '#0284c7',
    dark: '#0c4a6e',
  },
  success: {
    light: '#86efac',
    main: '#16a34a',
    dark: '#14532d',
  },
  error: {
    light: '#fca5a5',
    main: '#dc2626',
    dark: '#7f1d1d',
  },
  danger: {
    light: '#fda4af',
    main: '#e11d48',
    dark: '#881337',
  },
  warning: {
    light: '#fdba74',
    main: '#ea580c',
    dark: '#7c2d12',
  },
  singleTone: {
    white: '#ffffff',
    black: '#000000',
    lightText: '#ffffff',
    darkText: '#27272a',
  },
  grey: {
    50: '#fafafa',
    100: '#f5f5f5',
    200: '#e5e5e5',
    300: '#d4d4d4',
    400: '#a3a3a3',
    500: '#737373',
    600: '#525252',
    700: '#404040',
    800: '##262626',
    900: '#171717',
  },
  text: {
    disabled: '#d4d4d4',
    secondary: '#525252',
    primary: '#171717',
  },
};

export const typography: ThemeTypography = {
  htmlFontSize: 16,
  fontSize: 14,
  fontWeightLight: 300,
  fontWeightRegular: 400,
  fontWeightMedium: 500,
  fontWeightBold: 700,
  fontFamily: 'inherit',
};

export const opacity: ThemeOpacity = {
  0: 0,
  5: 0.05,
  10: 0.1,
  20: 0.2,
  25: 0.25,
  30: 0.3,
  40: 0.4,
  50: 0.5,
  60: 0.6,
  70: 0.7,
  75: 0.75,
  80: 0.8,
  90: 0.9,
  95: 0.95,
  100: 1,
};

export const shadows: ThemeShadows = {
  0: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },
  1: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 1.41,
    elevation: 2,
  },
  2: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
  },
  3: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    elevation: 4,
  },
  4: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  5: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.27,
    shadowRadius: 4.65,
    elevation: 6,
  },
  6: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.29,
    shadowRadius: 4.65,
    elevation: 7,
  },
  7: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    elevation: 8,
  },
  8: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.32,
    shadowRadius: 5.46,
    elevation: 9,
  },
  9: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,
  },
};

export const defaultTheme: LibraryDefaultTheme = {
  colors,
  typography,
  opacity,
  shadows,
};
