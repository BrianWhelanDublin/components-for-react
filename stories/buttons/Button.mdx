import { Story, Props, Source, Canvas } from '@storybook/addon-docs';

import Button from '../../src/components/buttons/elements/Button';

# Button

Buttons are clickable elements used to trigger actions. The button labels express what will occur when the user interacts with the button.

# Default Button

The default button is outlined with curved edges and a primary colour

<Canvas>
  <Story id="components-button--default" />
</Canvas>

# Square Button

The Square button add square edges to the buttons border

<Canvas>
  <Story id="components-button--square" />
</Canvas>

# Pill Button

The pill button add rounded edges to the button

<Canvas>
  <Story id="components-button--pill" />
</Canvas>

# Filled Button

The filled button adds a solid background to the button

<Canvas>
  <Story id="components-button--filled" />
</Canvas>

# Disabled Button

<Canvas>
  <Story id="components-button--disabled" />
</Canvas>

# Sizes

The button is available in three sizes small, medium and large. The default size is medium

<Canvas>
  <Story id="components-button--sizes" />
</Canvas>

## Button Colours

The button is available in the different theme colours

<Canvas>
  <Story id="components-button--colors" />
</Canvas>

## Component API

<Props />

Additional props passed into `Button` will be forwarded along to the underlying
button element.
