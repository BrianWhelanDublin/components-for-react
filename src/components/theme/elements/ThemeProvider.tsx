import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';
import { defaultTheme } from './theme';
import merge from 'deepmerge';

interface ComponentsThemeProviderProps {
  theme: DefaultTheme;
  children?: React.ReactNode;
}

const ComponentsThemeProvider: React.FC<ComponentsThemeProviderProps> = ({
  theme = {},
  children,
}) => {
  return (
    <ThemeProvider theme={merge(defaultTheme, theme)}>{children}</ThemeProvider>
  );
};

export default ComponentsThemeProvider;
