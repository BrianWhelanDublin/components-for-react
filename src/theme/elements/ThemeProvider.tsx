import React from 'react';
import { ThemeProvider } from 'styled-components';
import { LibraryDefaultTheme } from '../types/theme.types';
import { defaultTheme } from './theme';

interface ComponentsThemeProviderProps {
  theme: LibraryDefaultTheme;
  children?: React.ReactNode;
}

const ComponentsThemeProvider: React.FC<ComponentsThemeProviderProps> = ({
  theme,
  children,
}) => {
  return (
    <ThemeProvider theme={Object.assign({}, defaultTheme, theme)}>
      {children}
    </ThemeProvider>
  );
};

export default ComponentsThemeProvider;
