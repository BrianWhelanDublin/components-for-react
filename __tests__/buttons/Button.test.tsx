import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { defaultTheme } from '../../src/components/theme/elements/theme';
import Button from '../../src/components/buttons/elements/Button';
import renderer from 'react-test-renderer';

describe('Running Test for Button', () => {
  test('it works', () => {
    const { container } = render(<Button>Click</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Danger', () => {
    const button = renderer
      .create(
        <Button filled color="danger">
          Click
        </Button>
      )
      .toJSON();

    expect(button).toHaveStyleRule(
      'background',
      defaultTheme.colors?.danger?.main
    );
  });
});
