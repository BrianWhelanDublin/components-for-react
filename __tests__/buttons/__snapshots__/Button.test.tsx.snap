// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Button it works 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid transparent;
  margin: 0;
  -webkit-transition: border-color 0.25s ease-in-out,box-shadow 0.1s ease-in-out, background-color 0.25s ease-in-out,color 0.25s ease-in-out;
  transition: border-color 0.25s ease-in-out,box-shadow 0.1s ease-in-out, background-color 0.25s ease-in-out,color 0.25s ease-in-out;
  border-radius: 5px;
  margin: 0;
  cursor: pointer;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: inherit;
  font-weight: 500;
  padding: 12px 16px;
  font-size: 18px;
  background: inherit;
  border-color: #0891B2;
  color: #171717;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: default;
  background: #fafafa;
  color: #d4d4d4;
  border: #d4d4d4;
}

.c0:hover:not([disabled]) {
  background: transparent;
  border-color: #164E63;
}

<button
  class="c0"
  type="button"
>
  Click
</button>
`;
